### PR TITLE
Created model for user videos and a route for user videos

### PR DESCRIPTION
**Pull request by jimmique-parsons**
# Description

To allow users to access all of their created videos we created a `findByUser` model and an associated route (shown below). We also created a `findIdByUsername` model and associated route (shown below).

Screen Shot 2020-06-02 at 8 14 07 PM

`http://localhost:5050/api/videos/user/:userId`

Screen Shot 2020-06-02 at 8 56 55 PM

`http://localhost:5050/api/users/username/:username`

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [x] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [x] Manually tested in Postman, the route returns a 200 code. 
# Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: api/api_router.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-const video_update = require('../videos/video_update');
-const videos = require('../videos/videos');
-const songs = require('../songs/songs');
+const router = require("express").Router();
+const restricted = require("../middleware/restricted_middleware");
+const authRouter = require("../auth/auth_router");
+const userRouter = require("../users/users_router");
+const video_update = require("../videos/video_update");
+const videos = require("../videos/videos");
+const songs = require("../songs/songs");
```
Comment (panamajake86) [inline api/api_router.js:7]: These damn single vs double quotes.  I think it was some weird linter in a Mac vs. PC thing.  Been a thing the whole project.

File: users/users_model.js [hunk 2]
```diff
@@ -5,8 +5,9 @@ module.exports = {
   find,
   findBy,
   findById,
+  findIdByUsername,
   update,
-  remove,
+  remove
```
Comment (panamajake86) [inline users/users_model.js:10]: More linter fun again.

File: users/users_router.js [hunk 3]
```diff
@@ -1,26 +1,36 @@
...
+router.get("/username/:username", async (req, res) => {
+  const { username } = req.params;
+
+  try {
+    const id = await Users.findIdByUsername(username);
+    res.json(id);
+  } catch (err) {
+    res.send({ message: "Try again later.", err });
+  }
+});
```
Comment (panamajake86) [inline users/users_router.js:34]: Very nice.  I like this addition, wish I would've just done it when I first wrote this stuff.  Thanks for helping me with this stuff.

File: users/users_router.js [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
-module.exports = router;
\ No newline at end of file
+router.get("/username/:username", async (req, res) => {
+  const { username } = req.params;
+
+  try {
+    const id = await Users.findIdByUsername(username);
+    res.json(id);
+  } catch (err) {
+    res.send({ message: "Try again later.", err });
```
Comment (jtwray) [inline users/users_router.js:32]: We need to get some status codes in here for
 - 200 success
 - 400  error in request
 - 500  error on server

```suggestion
    res.status(500).send({ message: "Try again later.", err });
```

**Pull request merged**